Buffs NukeOp Shuttle into an avatar of death

Robustin opened the pull request:
:cl: Robustin
balance: The NukeOp Shuttle hull has been dramatically hardened. The walls are now "R-Walls" with far greater explosion resistance.
balance: The NukeOp Shuttle guns have been significantly buffed. They now rapidly fire a new type of penetrator round, at a greater range, and have far greater explosion resistance. 
balance: The nuclear device on the NukeOp Shuttle is now in an anchored state by default, the Nuke can only be unanchored by inserting the disk and entering the proper code. 
balance: Non-Syndicate cyborgs are now unable to access the NukeOp Shuttle Console.
fix: You can now unanchor Nukes even when the floor under them has been destroyed
/:cl:

This is intended to finally end the many years of incoherent NukeOp shuttle policy and other code changes designed to prevent detection of the Syndie shuttle. Not only will the shuttle be extremely dangerous to approach now, but even a breach of the shuttle is no longer a guarantee of victory since moving the ship will require an Agent ID card and moving the Nuke will require the disk.

Robustin committed:
Hardened Syndie Shuttle

Robustin committed:
Fixes

Robustin commented:
Ugh there's a mapmerge problem with the ship it seems, works fine on my end but I can't get the tool to convert the map because its missing a module or something and I can't install requirements because it wants Visual Studio C++ 14.0 even though I've downloaded the latest Visual Studio build tools, fucking map merge.

imsxz commented:
About time, after this we can get rid of the dumb shuttle hunting rule.

Anonmare commented:
Wouldn't it have made more sense to give Plastitanium the longer decon time r-walls have instead of making a snowflake-y subtype?

Just irks me

Robustin commented:
I mean these are just reskinned r-walls with more explosion block, it would take much more "snowflake" to bring over the dozen+ special rules that r-walls have over to a regular wall subtype.

Wilchenx commented:
plastitanium walls should just be made into better reinforced walls

PKPenguin321 commented:
you should let borgs still be able to use the console, but force them to be adjacent. maybe they should have to hack it first or something. that way the crew can still reverse-nuke the ops base

GunHog commented:
Can these walls be deconstructed by same way as normal R-walls? If so, did you test it for any weird sprite issues during the deconstruction process?

optimumtact commented:
make the doors hardened so theyr'e harder to hack too

optimumtact commented:
@GunHog they're just r-walls with more explosion resist

subject217 commented:
one minor concern here is that the ops will simply park the shuttle as close as they can get it in order to take advantage of the extremely powerful guns, and that they won't see much repercussion because the ship is very durable and explosion resistant. i'd definitely like to see how this plays out though.

MMMiracles commented:
RAMMING SPEED

Robustin commented:
@GunHog 

I blew up the shuttle a few times and manually deconstructed a few walls, I only encountered one icon issue that I fixed.

@subject217 

I'd honestly love to see more of that but keep in mind that these turrets still need "vision" to shoot. If turret can't "see" you, it won't target you. There's really only a couple places on station where you can get the guns close enough and still have enough windows to target crew, and the counter-strategy for that would simply be "don't go there if you don't have a plan for dealing with the guns", which already seems like it should be common sense. I gave the turret projectiles a highly limited range of 16 (a little over 1 screen length) so that the entire station doesn't get sprayed by the turrets either. ALSO the projectiles have much more noticeable icons so while their DPS is higher you don't have the obnoxious issue where if you approach the shuttle from the wrong angle you get insta-crit by 2 black pixels in space.

@PKPenguin321 

I'm not really willing to do that until there's an easier way to make the ship more borg-proof. The ship's blast door is already pretty lame because any Borg can use it and preventing that requires me to make a new Syndicate button subtype. Even if I addressed that, Engi borgs remain one of the best options for trying to board a syndicate vessel, I'm not comfortable with letting them GG the operatives just because they got into the cockpit. It's plausible to do something more elaborate but I don't have THAT much free time yet.

subject217 commented:
16 range, as in 16 turfs? you realize that your mob sits in the center of your screen, so your view range is 7 turfs, right?

SgtxRootbeer commented:
@Robustin the problem is that if the turrets' "vision" is their range, then they can hit you outside of your view, as whilst the projectile will travel an entire screen's distance, the player can only see half that distance in any one direction.

i remember playing a chief engineer once and as i was setting up solar panels, i was just south of xenobio and got sniped by turrets outside of my view, which was 10/10 gameplay. (it wasn't)

subject217 commented:
he gave them a scan_range of 9 so it's not a problem, just confusing wording in discussion.

SgtxRootbeer commented:
so its still gonna hit you from outside view range, gotcha.

Robustin commented:
The bullets are now far more visible and do 25 damage per hit. If you have a jetpack/gravity you can easily change course and survive.

SgtxRootbeer commented:
is it too much to ask that you see the turret that is shooting you, rather than being hit from offscreen?

Robustin commented:
Probably. It's been a common tactic to just sit off-screen of the shuttle, step forward, take a shot, step back and 2 turfs left or right, then rinse and repeat for low-risk turret removal.

Tlaltecuhtli commented:
can u test if  a drifted ttv can disable enough defences to make the turrets worthless? also will this affect the steelrain?

Repukan commented:
>Are these walls immune to engieborg rcd?

Robustin commented:
They're reskinned r_walls for the 5th time, so yes, RCDs cannot eat r_walls.

Anonmare commented:
@Robustin Engie borg RCDs are snowflake and can decon R-walls

Under line 495 in the game/objects/items/RCD.dm path, canRturf is set to true. Not sure if you care about that or not but letting you know

Robustin commented:
I didn't know, is that an upgrade? Did someone really decide that Engiborgs needed even MORE utility than their human counterpart?

Repukan commented:
Available roundstart, that's why I asked you three times

Robustin committed:
Fuck borgs

Robustin commented:
@WhoeverIsHeadminsNow

This is testmerge ready now. Mapmerge was still shitting itself on my home PC but I think dmm2tgm did the trick and it works fine locally.

ATH1909 commented:
Why shut down a valid anti-nukeop (anti-stallop/longop especially) strategy? While I'm okay with buffing the turrets and walls, the console should still be usable by anyone (even without an agent ID).

If the ops are worried about getting their shuttle stolen, they should hide it better/further away from the station.

Bawhoppen commented:
This really seems like a lame and ham-fisted way to deal with something that wasn't a huge problem in the first place. Making stuff arbitrarily more powerful, is not interesting design.

Robustin commented:
The console always required an Agent ID, borgs were the only ones who could just zip in and send the shuttle home making it impossible for the Ops to win. 

I don't want shuttle hunting to be the "best" strategy available, but if Borgs can offer 100% guaranteed victory just by reaching the cockpit then every Warops round will be someone with a jetpack, maxcaps, and dragging a borg behind them. It only seems fair that you shouldn't have a GUARANTEED victory over the Operatives without ever confronting them. 

@Bawhoppen 

Nothing ham-fisted about this. This PR makes the Syndicate shuttle into what it should have always been. These are Nuclear Operatives, station destroyers, using top-of-the-line combat tech, it has always been a joke that their ship was easier to break into than most of the station.

ShizCalev commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -679,6 +679,26 @@
+		addtimer(CALLBACK(src, .proc/shootAt, target), 5)
+		addtimer(CALLBACK(src, .proc/shootAt, target), 10)
+		addtimer(CALLBACK(src, .proc/shootAt, target), 15)
+		return 1
```
TRUE

ShizCalev commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -679,6 +679,26 @@
+		addtimer(CALLBACK(src, .proc/shootAt, target), 10)
+		addtimer(CALLBACK(src, .proc/shootAt, target), 15)
+		return 1
+	return
```
trailing return

Robustin committed:
Review

QualityVan commented:
let warops park their new battleship next to the station

GunHog commented:
> let warops park their new battleship next to the station

The flight computer lets you pick any open spot in space to move to ship, so you certainly can drop right next to the HoS' office and shoot it up!

ExcessiveUseOfCobblestone commented:
> If the ops are worried about getting their shuttle stolen, they should hide it better/further away from the station.

That's exactly why there will never be a fair playing field for people who want to shuttle hunt because they know OOCly that the shuttle cannot mechanically move anywhere but within the realm of the station Z level. If you go outside that Z level (which you can tell if you played enough), you know it's **100%** not there UNLESS someone moves it to the Z-level edge after, which also requires 1 person babysitting the shuttle and then promptly relaying that message to the team, hoping that no one has had their headset stolen during this time. That can be somewhat meta'd by looking at the server pop.

Shuttlehunting is forcing an unintended fight for the sole sake of denying the victory (hiding the nuke) to operatives without actually combating the antagonists. That said, I think this in tandem with the ability to purchase a spare nuke via TC would actually be okay since shuttlehunting wouldn't be a surefire way to deny victory at a cost of operatives losing out on some gear options.

ShizCalev commented:
Since we don't have mapdiffbot atm, can we get a pic?

Robustin commented:
@ShizCalev 

It's literally identical in appearance.

ShizCalev commented:
old
![ss13](https://user-images.githubusercontent.com/6209658/47389443-c15f4780-d6e2-11e8-8ae6-a901158c6c62.png)

new
![ss14](https://user-images.githubusercontent.com/6209658/47389449-c623fb80-d6e2-11e8-9755-d05964844131.png)

>floor patterns are now asymmetrical, all the turrets lost their preset directions (visual flare, but still), removed 1 O2 tank.

Exactly how it currently looks like.

ShizCalev reviewed (changes requested):
put the turret directions back in and make the floor pattern symmetrical again, please.

Robustin committed:
Syndie turrets are now set to lethal by default

nervere commented:
Rule 2, precedent 5 will be removed from /tg/ rules to accommodate if this PR gets merged.
This is what the rule reads:
`Station crew are currently not allowed to board or search for the syndicate shuttle during nuke op rounds, unless they specifically follow an operative or a pinpointer onto the shuttle. It is better to adminhelp first in this situation, so admins can confirm that you found it legitimately.`

ExcessiveUseOfCobblestone commented:
If oldman is going to keep this shtick up where he's just rude can he please make his biannual returns much shorter?

ShizCalev commented:
@Robustin

subject217 commented:
Now that's tragic. I'll have to take a look at getting this merged if Robustin has disappeared again.
